lib/assert.js

```diff
@@ -30,45 +30,49 @@ function test(ok, opts) {
 	}
 }
 
+function stringMsg(msg) {
+	return msg instanceof Error ? msg.message : typeof msg === "string" ? msg : msg ? JSON.stringify(msg) : null;
```

Comment:
It should fail if `msg` is anything other than a string, not gracefully handle it.